sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/PartitionProcessorImpl.java

```diff
@@ -124,18 +124,21 @@ public PartitionProcessorImpl(ChangeFeedObserver observer, ChangeFeedContextClie
 
                     CosmosClientException clientException = (CosmosClientException) throwable;
                     logger.warn("CosmosClientException: partition {} from thread {}",
-                        this.options.partitionKey().getInternalPartitionKey(), Thread.currentThread().getId(), clientException);
+                        this.settings.getPartitionKeyRangeId(), Thread.currentThread().getId(), clientException);
                     StatusCodeErrorType docDbError = ExceptionClassifier.classifyClientException(clientException);
 
                     switch (docDbError) {
                         case PARTITION_NOT_FOUND: {
                             this.resultException = new PartitionNotFoundException("Partition not found.", this.lastContinuation);
+                            break;
                         }
                         case PARTITION_SPLIT: {
                             this.resultException = new PartitionSplitException("Partition split.", this.lastContinuation);
```

Comment:
@milismsft we need to have explicit automated tests for partition split.

We have some tests for partition split that you can use as a sample:
DocumentProducerTest (unit test)
ReadMyWritesConsistencyTest (integration test)

Reply:
I've added a specific test for splits in V4 respective change. The challenge is that public emulator does not support splitting so the test must be run against the cloud service.

Reply:
if we add tests on v4 that should be good enough. thanks.